kratos/geometries/nurbs_curve_geometry.h

```diff
@@ -0,0 +1,274 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Thomas Oberbichler
+//
+//  Ported from the ANurbs library (https://github.com/oberbichler/ANurbs)
+//
+
+#if !defined(KRATOS_NURBS_CURVE_GEOMETRY_H_INCLUDED )
+#define  KRATOS_NURBS_CURVE_GEOMETRY_H_INCLUDED
+
+#include "nurbs_curve_shape_functions.h"
+#include "nurbs_interval.h"
+#include "includes/ublas_interface.h"
+#include "containers/array_1d.h"
+
+#include <stdexcept>
+#include <vector>
+
+namespace Kratos {
+
+template <int TDimension, typename TPoleContainer =
+    std::vector<array_1d<double, TDimension>>>
+class NurbsCurveGeometry
+{
+public:     // types
+    using Type = NurbsCurveGeometry<TDimension>;
+    using Vector = array_1d<double, TDimension>;
+
+private:    // variables
+    const int mDegree;
+    std::vector<double> mKnots;
+    TPoleContainer mPoles;
+    std::vector<double> mWeights;
+
+public:     // constructors
+    NurbsCurveGeometry(
+        const int Degree,
+        int NbPoles,
+        bool IsRational
+    ) : mDegree(Degree),
+        mPoles(NbPoles),
+        mWeights(IsRational ? NbPoles : 0),
+        mKnots(NurbsUtility::GetNbKnots(Degree, NbPoles))
+    {
+        static_assert(TDimension > 0, "Invalid dimension");
+    }
+
+    NurbsCurveGeometry(
+        const int Degree,
+        const std::vector<double>& rKnots,
```

Comment:
can this be a Vector also?

Reply:
Yes. But it is actually just a list of doubles and not a vector like in linear algebra.

Reply:
Yes, but I am talking about the Kratos Vector.